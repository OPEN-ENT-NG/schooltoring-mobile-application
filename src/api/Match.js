import axios from "axios";

async function getMatches(state, page) {
  try {
    const request = {
      method: "GET",
      url: `${
        global.config.auth.endpoint
      }/schooltoring/match/${state}?page=${page}`
    };
    const { data } = await axios(request);
    return data;
  } catch (err) {
    throw err;
  }
}

export default {
  getMatches
};
