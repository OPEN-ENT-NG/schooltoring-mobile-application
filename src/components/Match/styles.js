import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const center = {
  flexDirection: "row",
  alignItems: "center",
  justifyContent: "center"
};

const roundButton = {
  width: 66,
  height: 66,
  borderRadius: 33,
  ...center
};

export default StyleSheet.create({
  detail: {
    backgroundColor: COLORS.SECONDARY_BACKGROUND,
    padding: 20
  },
  center,
  marginVertical: {
    marginVertical: 15
  },
  roundButton,
  declineButton: {
    ...roundButton,
    backgroundColor: COLORS.SECONDARY
  },
  acceptButton: {
    ...roundButton,
    backgroundColor: COLORS.PRIMARY
  },
  roundButtonIcon: {
    fontSize: 30,
    color: COLORS.WHITE
  },
  favoriteButton: {
    fontSize: 24,
    color: COLORS.GREY
  },
  height30: {
    height: "30%"
  },
  height40: {
    height: "40%"
  },
  width25: {
    width: "25%"
  }
});
