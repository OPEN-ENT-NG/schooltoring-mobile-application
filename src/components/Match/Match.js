import React, { Component } from "react";
import { View } from "react-native";
import Icon from "react-native-vector-icons/MaterialIcons";

import Proptypes from "prop-types";

import Identity from "../Identity/Identity";
import SubjectBadge from "../SubjectBadge/SubjectBadge";
import DayBadge from "../DayBadge/DayBadge";
import SecondaryButton from "../SecondaryButton/SecondaryButton";
import Touchable from "../Touchable/Touchable";

import I18n from "../../api/I18n";

import styles from "./styles";
import { COLORS } from "../../styles/common";

class Match extends Component {
  renderAvailabilites() {
    let days = Object.keys(this.props.availabilities);
    let render = [];
    days.forEach(day =>
      render.push(
        <DayBadge
          key={day}
          style={{ marginHorizontal: 3 }}
          label={I18n.t(`availability.${day}`)
            .charAt(0)
            .toUpperCase()}
          available={this.props.availabilities[day]}
        />
      )
    );
    return render;
  }

  render() {
    return (
      <View>
        <View style={[styles.height30, styles.center]}>
          <Touchable
            style={styles.center}
            onPress={this.props.viewProfile}
            hitSlop={{ top: 10, left: 10, bottom: 10, right: 10 }}>
            <Identity
              userinfo={this.props.userinfo}
              avatar={{ size: 100, src: this.props.userinfo.avatar }}
            />
          </Touchable>
        </View>
        <View
          style={[
            styles.detail,
            styles.height40,
            { justifyContent: "space-evenly" }
          ]}>
          <View
            style={[
              styles.center,
              styles.marginVertical,
              { marginBottom: 40, flexWrap: "wrap" }
            ]}>
            {this.props.features.map(feature => (
              <SubjectBadge
                key={feature.subjectId}
                title={feature.subjectLabel}
                color={
                  this.props.state.toUpperCase() === "STRENGTH"
                    ? COLORS.SECONDARY
                    : COLORS.PRIMARY
                }
              />
            ))}
          </View>
          <View style={[styles.center, styles.marginVertical]}>
            {this.renderAvailabilites()}
          </View>
        </View>
        <View style={[styles.height30, styles.center]}>
          <View
            style={[
              styles.center,
              {
                flex: 1,
                justifyContent: "flex-end"
              }
            ]}>
            <View style={[styles.width25, styles.center]}>
              <SecondaryButton
                onPress={this.props.onClear}
                style={styles.declineButton}>
                <Icon name="clear" style={styles.roundButtonIcon} />
              </SecondaryButton>
            </View>
            <View style={[styles.width25, styles.center]}>
              <SecondaryButton
                onPress={this.props.onChat}
                style={styles.acceptButton}>
                <Icon name="send" style={styles.roundButtonIcon} />
              </SecondaryButton>
            </View>
            <View style={[styles.width25, styles.center]}>
              <SecondaryButton
                onPress={this.props.onFavorite}
                style={styles.roundButton}>
                <Icon
                  name="favorite"
                  color={this.props.isFavorite ? "red" : COLORS.GREY}
                  style={styles.favoriteButton}
                />
              </SecondaryButton>
            </View>
          </View>
        </View>
      </View>
    );
  }
}

Match.proptypes = {
  userinfo: Proptypes.shape({
    username: Proptypes.string.isRequired,
    avatar: Proptypes.string.isRequired,
    classNames: Proptypes.array.isRequired
  }),
  features: Proptypes.array.isRequired,
  availabilities: Proptypes.shape({
    monday: Proptypes.bool.isRequired,
    tuesday: Proptypes.bool.isRequired,
    wednesday: Proptypes.bool.isRequired,
    thursday: Proptypes.bool.isRequired,
    friday: Proptypes.bool.isRequired,
    saturday: Proptypes.bool.isRequired,
    sunday: Proptypes.bool.isRequired
  }).isRequired,
  state: Proptypes.string.isRequired,
  onClear: Proptypes.func.isRequired,
  onChat: Proptypes.func.isRequired,
  onFavorite: Proptypes.func.isRequired,
  isFavorite: Proptypes.bool,
  viewProfile: Proptypes.func.isRequired
};

export default Match;
