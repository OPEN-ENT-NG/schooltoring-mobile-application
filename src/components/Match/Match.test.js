import React from "react";
import { shallow } from "enzyme";

import Match from "./Match";

const match = {
  id: "49abe348-0b13-4ce0-ac8c-01fa7b7e8f68",
  matches: 3,
  availabilities: {
    monday: true,
    tuesday: false,
    wednesday: false,
    thursday: false,
    friday: false,
    saturday: false,
    sunday: false
  },
  features: [
    { subjectId: "1474925-1517929209848", subjectLabel: "FRANCAIS" },
    { subjectId: "1480091-1517929209848", subjectLabel: "ALLEMAND LV2" },
    { subjectId: "1474895-1517929209848", subjectLabel: "ANGLAIS LV1" }
  ],
  userinfo: {
    id: "49abe348-0b13-4ce0-ac8c-01fa7b7e8f68",
    username: "Amel AMAIRI",
    classNames: ["VERSAILLES-7133$BTS1TPIL"],
    avatar: "/workspace/document/916c2df5-13d2-48cf-9f54-4402c36d4af0"
  }
};

const onClear = jest.fn();
const onChat = jest.fn();
const onFavorite = jest.fn();

const renderedStrengths = shallow(
  <Match
    userinfo={match.userinfo}
    features={match.features}
    availabilities={match.availabilities}
    state="STRENGTH"
    onClear={onClear}
    onChat={onChat}
    onFavorite={onFavorite}
  />
);

const renderedWkeanesses = shallow(
  <Match
    userinfo={match.userinfo}
    features={match.features}
    availabilities={match.availabilities}
    state="WEAKNESS"
    onClear={onClear}
    onChat={onChat}
    onFavorite={onFavorite}
  />
);

describe("Render", () => {
  test("Strength match should render without crashing", () =>
    expect(renderedStrengths).toMatchSnapshot());

  test("Weaknesse match should render without crashing", () =>
    expect(renderedWkeanesses).toMatchSnapshot());

  test("Match should render 3 features", () =>
    expect(renderedStrengths.find("SubjectBadge").length).toEqual(3));

  test("Match shoould render 'FRANCAIS', 'ALLEMAND LV2' and 'ANGLAIS LV1' as features", () => {
    const features = renderedStrengths.find("SubjectBadge");
    expect(
      features
        .at(0)
        .dive()
        .find("Text")
        .contains("FRANCAIS")
    ).toBeTruthy();
    expect(
      features
        .at(1)
        .dive()
        .find("Text")
        .contains("ALLEMAND LV2")
    ).toBeTruthy();
    expect(
      features
        .at(2)
        .dive()
        .find("Text")
        .contains("ANGLAIS LV1")
    ).toBeTruthy();
  });

  test("Match should render 7 availabilities", () =>
    expect(renderedStrengths.find("DayBadge").length).toEqual(7));

  test("Match should render 3 SecondaryButtons", () =>
    expect(renderedStrengths.find("SecondaryButton").length).toEqual(3));

  test("Match should render 3 buttons with respectively 'clear', 'chat' and 'favorite' as Icon", () => {
    const buttons = renderedStrengths.find("SecondaryButton");

    expect(
      buttons
        .at(0)
        .children()
        .find("Icon")
        .props().name
    ).toEqual("clear");
    expect(
      buttons
        .at(1)
        .children()
        .find("Icon")
        .props().name
    ).toEqual("chat");
    expect(
      buttons
        .at(2)
        .children()
        .find("Icon")
        .props().name
    ).toEqual("favorite");
  });
});

describe("Interactions", () => {
  test("On press on clear button should call onClear prop function", () => {
    onClear.mockReset();
    const onClearButton = renderedStrengths.find("SecondaryButton").at(0);
    onClearButton.props().onPress();

    expect(onClear).toHaveBeenCalled();
  });

  test("On press on chat button should call onChat prop function", () => {
    onChat.mockReset();
    const onClearButton = renderedStrengths.find("SecondaryButton").at(1);
    onClearButton.props().onPress();

    expect(onChat).toHaveBeenCalled();
  });

  test("On press on favorite button should call onClear prop function", () => {
    onFavorite.mockReset();
    const onClearButton = renderedStrengths.find("SecondaryButton").at(2);
    onClearButton.props().onPress();

    expect(onFavorite).toHaveBeenCalled();
  });
});
