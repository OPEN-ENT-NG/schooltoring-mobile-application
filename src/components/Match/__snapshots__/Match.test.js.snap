// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Strength match should render without crashing 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "height": "30%",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Touchable
      hitSlop={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Identity
        avatar={
          Object {
            "size": 100,
            "src": "/workspace/document/916c2df5-13d2-48cf-9f54-4402c36d4af0",
          }
        }
        userinfo={
          Object {
            "avatar": "/workspace/document/916c2df5-13d2-48cf-9f54-4402c36d4af0",
            "classNames": Array [
              "VERSAILLES-7133$BTS1TPIL",
            ],
            "id": "49abe348-0b13-4ce0-ac8c-01fa7b7e8f68",
            "username": "Amel AMAIRI",
          }
        }
      />
    </Touchable>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e3f2fd",
          "padding": 20,
        },
        Object {
          "height": "40%",
        },
        Object {
          "justifyContent": "space-evenly",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "marginVertical": 15,
          },
          Object {
            "flexWrap": "wrap",
            "marginBottom": 40,
          },
        ]
      }
    >
      <SubjectBadge
        color="#e91e63"
        key="1474925-1517929209848"
        title="FRANCAIS"
      />
      <SubjectBadge
        color="#e91e63"
        key="1480091-1517929209848"
        title="ALLEMAND LV2"
      />
      <SubjectBadge
        color="#e91e63"
        key="1474895-1517929209848"
        title="ANGLAIS LV1"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "marginVertical": 15,
          },
        ]
      }
    >
      <DayBadge
        available={true}
        key="monday"
        label="L"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="tuesday"
        label="M"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="wednesday"
        label="M"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="thursday"
        label="J"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="friday"
        label="V"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="saturday"
        label="S"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="sunday"
        label="D"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": "30%",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "25%",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <SecondaryButton
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#e91e63",
              "borderRadius": 33,
              "flexDirection": "row",
              "height": 66,
              "justifyContent": "center",
              "width": 66,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="clear"
            size={12}
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 30,
              }
            }
          />
        </SecondaryButton>
      </View>
      <View
        style={
          Array [
            Object {
              "width": "25%",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <SecondaryButton
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#17b6b0",
              "borderRadius": 33,
              "flexDirection": "row",
              "height": 66,
              "justifyContent": "center",
              "width": 66,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="send"
            size={12}
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 30,
              }
            }
          />
        </SecondaryButton>
      </View>
      <View
        style={
          Array [
            Object {
              "width": "25%",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <SecondaryButton
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 33,
              "flexDirection": "row",
              "height": 66,
              "justifyContent": "center",
              "width": 66,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#929292"
            name="favorite"
            size={12}
            style={
              Object {
                "fontSize": 24,
              }
            }
          />
        </SecondaryButton>
      </View>
    </View>
  </View>
</View>
`;

exports[`Render Weaknesse match should render without crashing 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "height": "30%",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Touchable
      hitSlop={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Identity
        avatar={
          Object {
            "size": 100,
            "src": "/workspace/document/916c2df5-13d2-48cf-9f54-4402c36d4af0",
          }
        }
        userinfo={
          Object {
            "avatar": "/workspace/document/916c2df5-13d2-48cf-9f54-4402c36d4af0",
            "classNames": Array [
              "VERSAILLES-7133$BTS1TPIL",
            ],
            "id": "49abe348-0b13-4ce0-ac8c-01fa7b7e8f68",
            "username": "Amel AMAIRI",
          }
        }
      />
    </Touchable>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e3f2fd",
          "padding": 20,
        },
        Object {
          "height": "40%",
        },
        Object {
          "justifyContent": "space-evenly",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "marginVertical": 15,
          },
          Object {
            "flexWrap": "wrap",
            "marginBottom": 40,
          },
        ]
      }
    >
      <SubjectBadge
        color="#17b6b0"
        key="1474925-1517929209848"
        title="FRANCAIS"
      />
      <SubjectBadge
        color="#17b6b0"
        key="1480091-1517929209848"
        title="ALLEMAND LV2"
      />
      <SubjectBadge
        color="#17b6b0"
        key="1474895-1517929209848"
        title="ANGLAIS LV1"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "marginVertical": 15,
          },
        ]
      }
    >
      <DayBadge
        available={true}
        key="monday"
        label="L"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="tuesday"
        label="M"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="wednesday"
        label="M"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="thursday"
        label="J"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="friday"
        label="V"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="saturday"
        label="S"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
      <DayBadge
        available={false}
        key="sunday"
        label="D"
        style={
          Object {
            "marginHorizontal": 3,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": "30%",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "25%",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <SecondaryButton
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#e91e63",
              "borderRadius": 33,
              "flexDirection": "row",
              "height": 66,
              "justifyContent": "center",
              "width": 66,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="clear"
            size={12}
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 30,
              }
            }
          />
        </SecondaryButton>
      </View>
      <View
        style={
          Array [
            Object {
              "width": "25%",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <SecondaryButton
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#17b6b0",
              "borderRadius": 33,
              "flexDirection": "row",
              "height": 66,
              "justifyContent": "center",
              "width": 66,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="send"
            size={12}
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 30,
              }
            }
          />
        </SecondaryButton>
      </View>
      <View
        style={
          Array [
            Object {
              "width": "25%",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <SecondaryButton
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 33,
              "flexDirection": "row",
              "height": 66,
              "justifyContent": "center",
              "width": 66,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#929292"
            name="favorite"
            size={12}
            style={
              Object {
                "fontSize": 24,
              }
            }
          />
        </SecondaryButton>
      </View>
    </View>
  </View>
</View>
`;
